src/poetry/console/commands/show.py

```diff
@@ -211,7 +211,7 @@ def _display_packages_information(
         with solver.use_environment(self.env):
             ops = solver.solve().calculate_operations()
 
-        required_locked_packages = {op.package for op in ops if not op.skipped}
+        required_locked_packages = {op.package.name for op in ops if not op.skipped}
```

Comment:
This change will introduce a bug that is unfortunately not discovered by our test suite. I just created #6920 to increase test coverage.